WcaOnRails/app/models/competition.rb

```diff
@@ -107,7 +108,11 @@ def delegates_must_be_delegates
     end
   end
 
-  def warnings
+  def user_should_post_delegate_report?(user)
+    !new_record? && is_over? && !delegate_report.posted? && delegates.include?(user)
+  end
```

Comment:
Ah darn. So that actually doesn't work because it breaks this validation in DelegateReport:

```ruby
  validate :only_post_after_competition
  def only_post_after_competition
    if posted? && !can_be_posted?
      errors.add(:posted, "cannot be posted yet")
    end
  end
```

I went with something inbetween. I'm pushing my latest change now.